### ISSUE TITLE
DynamoDB: Unable to put binary data

### ISSUE THREAD

**Issue by der-flo** [aws/aws-sdk-ruby#831]
When trying to put binary data to DynamoDB, I get an exception regarding the usage of an IO (eg. `StringIO`) type. I'm using the actual gem (2.0.47).

Simple example to reproduce:
``` ruby
require 'aws-sdk'
client = Aws::DynamoDB::Client.new(region: 'eu-west-1')

item = {
  'id' => 'bar',
  'foo' => StringIO.new('baz')
}
client.put_item(table_name: 'foobar', item: item)
```

Exception and backtrace:
```
[...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:54:in `format': undefined method `read' for "baz":String (NoMethodError)
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:24:in `block in structure'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:20:in `each'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:20:in `with_object'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:20:in `structure'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:50:in `format'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:41:in `block in map'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:40:in `each'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:40:in `with_object'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:40:in `map'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:52:in `format'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:24:in `block in structure'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:20:in `each'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:20:in `with_object'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:20:in `structure'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:50:in `format'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/builder.rb:11:in `to_json'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/rpc_body_handler.rb:20:in `build_json'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/rpc_body_handler.rb:8:in `call'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/json/rpc_headers_handler.rb:10:in `call'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/plugins/restful_bindings.rb:13:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/plugins/endpoint.rb:31:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/plugins/param_validation.rb:22:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/plugins/dynamodb_simple_attributes.rb:112:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/plugins/param_conversion.rb:22:in `call'
from [...]/aws-sdk-core-2.0.47/lib/aws-sdk-core/plugins/response_paging.rb:10:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/plugins/response_target.rb:18:in `call'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/request.rb:70:in `send_request'
from [...]/aws-sdk-core-2.0.47/lib/seahorse/client/base.rb:216:in `block (2 levels) in define_operation_methods'

**Comment by awood45** [comment]
Thanks for this, I was able to reproduce and I'm taking a look.

**Comment by awood45** [comment]
I have a possible fix written which passes in your case, but there are a few other cases I want to check in review. Will push up my changes to a branch and PR.

**Comment by trevorrowe** [comment]
Closed by #832

**Comment by der-flo** [comment]
Thanks for version 2.0.48!
